### PR TITLE
Join messages

### PR DESCRIPTION
**Pull request by vfurmane**

### CODE REVIEW

**Review by ppiques — APPROVED** [review]
TL; still DR

File: include/IRCPeerManager.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		void	add(int fd, struct sockaddr &addr);
+		void	remove(int fd);
+		TCPPeer	&get(int fd);
+		int		acceptConnection(void);
+		void	closeConnection(int fd);
+
+	private:
+		TCPServer				&_server;
+		std::map<int, TCPPeer> _peers;
+};
```
Comment (theovgl) [inline include/IRCPeerManager.hpp:30]: Wrong class name :(

File: src/IRCPeerManager.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+void	TCPPeerManager::closeConnection(int fd)
+{
+#ifndef NDEBUG
+	std::cerr << "Closing connection on fd no " << fd << "..." << std::endl;
+	std::cerr << "IP address -> " << this->get(fd).getStrAddr() << std::endl;
+#endif
+	if (this->_peers.at(fd).close() == -1)
+		throw sysCallError("close", strerror(errno));
+	this->remove(fd);
+}
```
Comment (theovgl) [inline src/IRCPeerManager.cpp:74]: Wrong class name again

**Review by theovgl — CHANGES_REQUESTED** [review]

**Review by ppiques — APPROVED** [review]
style

**Review by ppiques — APPROVED** [review]

**Review by theovgl — APPROVED** [review]

**Review by ppiques — APPROVED** [review]

**Pull request merged**